perfkitbenchmarker/providers/azure/azure_network.py

```diff
@@ -21,264 +21,284 @@
 from perfkitbenchmarker import resource
 from perfkitbenchmarker import vm_util
 from perfkitbenchmarker import providers
+from perfkitbenchmarker.providers import azure
 
 FLAGS = flags.FLAGS
-AZURE_PATH = 'azure'
-MAX_NAME_LENGTH = 24
 SSH_PORT = 22
-# We need to prefix storage account names so that VMs won't create their own
-# account upon creation.
-# See https://github.com/MSOpenTech/azure-xplat-cli/pull/349
-STORAGE_ACCOUNT_PREFIX = 'portalvhds'
 
+DEFAULT_LOCATION = 'eastus2'
 
-class _AzureEndpoint(resource.BaseResource):
-  """An object representing an endpoint to an Azure VM.
 
-  No deletion is specified, as endpoints are deleted along with the VM.
-  """
-  def __init__(self, vm_name, port, protocol):
-    super(_AzureEndpoint, self).__init__()
-    self.vm_name = vm_name
-    self.port = port
-    self.protocol = protocol
-
-  def _Create(self):
-    create_cmd = [AZURE_PATH,
-                  'vm',
-                  'endpoint',
-                  'create',
-                  self.vm_name,
-                  str(self.port),
-                  '--protocol=' + self.protocol]
-    vm_util.IssueCommand(create_cmd)
-
-  def _Exists(self):
-    """Returns whether or not an endpoint exists."""
-    # Example output:
-    # [
-    #   {
-    #     "localPort": 22,
-    #     "name": "ssh",
-    #     "port": 22,
-    #     "protocol": "tcp",
-    #     "virtualIPAddress": "104.43.224.13",
-    #     "enableDirectServerReturn": false
-    #   }
-    # ]
-    exists_cmd = [AZURE_PATH,
-                  'vm',
-                  'endpoint',
-                  'list',
-                  '--json',
-                  self.vm_name]
-    stdout, _, status = vm_util.IssueCommand(exists_cmd)
-    if status or stdout == 'No VMs found':
-      return False
-    else:
-      arr = json.loads(stdout)
-      return any(ep['port'] == self.port and ep['protocol'] == self.protocol
-                 for ep in arr)
-
-  def _Delete(self):
-    """Endpoint will be deleted with VM, so this is a noop."""
-    pass
-
-
-class AzureFirewall(network.BaseFirewall):
-  """An object representing the Azure Firewall equivalent.
-
-  On Azure, endpoints are used to open ports instead of firewalls.
-  """
-
-  CLOUD = providers.AZURE
-
-  def AllowPort(self, vm, port):
-    """Opens a port on the firewall.
-
-    Args:
-      vm: The BaseVirtualMachine object to open the port for.
-      port: The local port to open.
-    """
-    if vm.is_static or port == SSH_PORT:
-      return
-    _AzureEndpoint(vm.name, port, 'tcp').Create()
-    _AzureEndpoint(vm.name, port, 'udp').Create()
-
-  def DisallowAllPorts(self):
-    """Closes all ports on the firewall."""
-    pass
+def GetResourceGroup():
+  """Get the resource group for the current benchmark."""
+  spec = context.GetThreadBenchmarkSpec()
+  if hasattr(spec, 'azure_resource_group'):
+    return spec.azure_resource_group
+  else:
+    group = AzureResourceGroup('pkb%s%s' % (FLAGS.run_uri, spec.uid))
```

Comment:
I did it this way because the name will be a prefix of the name of the storage account, and storage account names can't have any characters except for lower-case letters and numbers. But no, I really want a hyphen. I'll go back and fix that up.